### ISSUE TITLE
False positive for discouraged_direct_init 

### ISSUE THREAD

**Issue by saviosailas** [realm/SwiftLint#3230]
### New Issue Checklist

- [x]  Updated SwiftLint to the latest version

- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

False positive for discouraged_direct_init when UIDevice extension variable is used. To access variable in extension UIDevice() is needed. Please remove this rule for variables in extension.

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
$ swiftlint lint checkDevice.swift
Loading configuration from '.swiftlint.yml'
Linting Swift files at paths checkDevice.swift
Linting 'checkDevice.swift' (1/1)
checkDevice.swift:18:22: warning: Discouraged Direct Initialization Violation: Discouraged direct initialization of types that can be harmful. (discouraged_direct_init)
Done linting! Found 1 violation, 0 serious in 1 file
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? - 0.39.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? - Homebrew
* Paste your configuration file:

```yml
# insert yaml contents here
disabled_rules:
- control_statement
excluded:
  - Pods/*/*.swift
  - Pods/*.swift
  - Pods
```

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)? - No
  If so, paste their relative paths and respective contents.
* Which Xcode version are you using (check `xcodebuild -version`)? - 11.3
* Do you have a sample that shows the issue? Run ` swiftlint lint checkDevice.swift`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
// This triggers a violation:
let deviceModel = UIDevice().deviceModelName.rawValue 

// Extension file:
public extension UIDevice {
       var deviceModelName: ModelName {
          //  Business logic
         // ModelName is enum
       }
}
```
